ripple/src/main/webapp/script.js

```diff
@@ -22,17 +29,65 @@ function fetchBlobstoreUploadUrl(formId, fileId, webUrl) {
       });
 }
 
+/* Serves blob by parsing blobKey from parameter. If no parameter, serves stored blob. */
+function getBlobKey(uid, blobKey) {
+  var newBlobKey = getParameterByName('blob-key');
+  console.log("Parameter blobKey: " + newBlobKey);
+  // New image uploaded. Store blobkey in firestore.
+  if (newBlobKey != null) {
+    console.log("New image. Blobkey parameter: ", newBlobKey);
+    db.collection("users").doc(uid).update({
+      blobKey: newBlobKey,
+    })
+        .then(function() {
```

Comment:
then() and catch() formatting seems off?

Reply:
Fixed